Index: Shared/src/org/bimserver/shared/ServiceInterface.java
===================================================================
--- Shared/src/org/bimserver/shared/ServiceInterface.java	(revision 889)
+++ Shared/src/org/bimserver/shared/ServiceInterface.java	(working copy)
@@ -68,13 +68,13 @@
 public interface ServiceInterface {
 	@GET
 	@Path("/ping")
-	@Produces("application/text")
+	@Produces("text/plain")
 	@WebMethod(action = "ping")
 	String ping(@QueryParam("in") @WebParam(name = "in", partName = "ping.in") String in);
 
 	@GET
 	@Path("/login")
-	@Produces("application/xml")
+	@Produces("text/plain")
 	@WebMethod(action = "login")
 	boolean login(@QueryParam("username") @WebParam(name = "username", partName = "login.username") String username,
 			@QueryParam("password") @WebParam(name = "password", partName = "login.password") String password) throws UserException,
@@ -110,12 +110,13 @@
 
 	@GET
 	@Path("/download")
-	@Produces("application/xml")
+	@Produces("text/plain")
 	@WebMethod(action = "download")
 	int download(@QueryParam("roid") @WebParam(name = "roid", partName = "download.roid") long roid,
 			@QueryParam("resultType") @WebParam(name = "formatIdentifier", partName = "download.formatIdentifier") String formatIdentifier,
 			@QueryParam("sync") @WebParam(name = "sync", partName = "download.sync") boolean sync) throws UserException, ServerException;
 
+
 	@WebMethod(action = "downloadByOids")
 	int downloadByOids(@WebParam(name = "roids", partName = "downloadByOids.roids") Set<Long> roids,
 			@WebParam(name = "oids", partName = "downloadByOids.oids") Set<Long> oids,
@@ -195,13 +196,13 @@
 
 	@GET
 	@Path("/getAllProjects")
-	@Produces("application/xml")
+	@Produces({"application/json", "application/xml"})
 	@WebMethod(action = "getAllProjects")
 	List<SProject> getAllProjects() throws UserException, ServerException;
 
 	@GET
 	@Path("/getAllReadableProjects")
-	@Produces("application/xml")
+	@Produces({"application/json", "application/xml"})
 	@WebMethod(action = "getAllReadableProjects")
 	List<SProject> getAllReadableProjects() throws UserException, ServerException;
 
@@ -210,7 +211,7 @@
 
 	@GET
 	@Path("/getAllRevisionsOfProject")
-	@Produces("application/xml")
+	@Produces({"application/json", "application/xml"})
 	@WebMethod(action = "getAllRevisionsOfProject")
 	List<SRevision> getAllRevisionsOfProject(
 			@QueryParam("poid") @WebParam(name = "poid", partName = "getAllRevisionsOfProject.poid") long poid) throws UserException,
@@ -299,9 +300,12 @@
 			@WebParam(name = "className", partName = "getDataObjectByOid.className") String className) throws UserException,
 			ServerException;
 
+    @GET
+    @Path("/getDataObjectByGuid")
+    @Produces({"application/json", "application/xml"})
 	@WebMethod(action = "getDataObjectByGuid")
-	SDataObject getDataObjectByGuid(@WebParam(name = "roid", partName = "getDataObjectByGuid.roid") long roid,
-			@WebParam(name = "guid", partName = "getDataObjectByGuid.guid") String guid) throws UserException, ServerException;
+	SDataObject getDataObjectByGuid(@QueryParam("roid") @WebParam(name = "roid", partName = "getDataObjectByGuid.roid") long roid,
+			@QueryParam("guid") @WebParam(name = "guid", partName = "getDataObjectByGuid.guid") String guid) throws UserException, ServerException;
 
 	@WebMethod(action = "getDataObjectsByType")
 	List<SDataObject> getDataObjectsByType(@WebParam(name = "roid", partName = "getDataObjectsByType.roid") long roid,
